stl/inc/xutility

```diff
@@ -3246,10 +3249,38 @@ namespace ranges {
         }
     };
 
-    // VARIABLE ranges::distance
     inline constexpr _Distance_fn distance{_Not_quite_object::_Construct_tag{}};
 
-    // CLASS ranges::_Next_fn
+    // VARIABLE ranges::ssize
+    class _Ssize_fn { // Per the proposed resolution of LWG-3403
+    public:
+        // clang-format off
+        template <class _Rng>
+        _NODISCARD constexpr auto operator()(_Rng&& _Range) const requires requires { _RANGES size(_Range); } {
+            using _SSSsssigned = _Make_signed_like_t<decltype(_RANGES size(_Range))>;

```

Comment:
What happens if `ranges::size` returns `const int&`? Do we care if it (even more pathologically) returns `bool`?

Reply:
`const int&` is off the table: the spec guarantees that `ranges::size(x)` is a prvalue of integer-like type. `bool`, however, is a problem. We should have excluded `bool` from the integer-like types. I need to file another LWG issue.

Reply:
Issue submitted.

Reply:
Should we also exclude the `char` family?

Reply:
That's a good question. It does seem like the character types would be wacky as range size types, although I'd worry about `char` colliding with `int8_t` or `uint8_t` on certain systems (not ours).

Reply:
While I agree it would be peculiar to choose `char`, `wchar_t`, or `charN_t` instead of `signed char`/`unsigned char` or the underlying type of `wchar_t`/`charN_t`, I see no particular reason to forbid them as integer-like types. I honestly wouldn't even mind `bool` were it not the case that we need `make_signed_t` and `make_unsigned_t` to work.